hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALProvider.java

```diff
@@ -130,7 +130,12 @@ void init(WALFactory factory, Configuration conf, String providerId, Abortable s
   void addWALActionsListener(WALActionsListener listener);
 
   default WALFileLengthProvider getWALFileLengthProvider() {
-    return path -> getWALs().stream().map(w -> w.getLogFileSizeIfBeingWritten(path))
-        .filter(o -> o.isPresent()).findAny().orElse(OptionalLong.empty());
+    return path -> getWALs().stream().map(w -> {
+      try {
+        return w.getLogFileSizeIfBeingWritten(path);
+      } catch (IOException e) {
+        return OptionalLong.empty();
```

Comment:
Is this safe? If we could always ignore this exception, let's just do not declare getLogFileSizeIfBeingWritten as throwing IOException?

Reply:
Thanks @Apache9  for reviewing.
In this PR, I introduce `RemoteWALFileLengthProvider` implementing WALFileLengthProvider, it is hold by ReplicationServer and gets the length of wals from RS through RPC,  it may throw an IOE. So I need declare `WALFileLengthProvider.getLogFileSizeIfBeingWritten` as throwing IOException.
But this is safe here, `WALProvider` is only used by RS, `getWALs` returns `WAL` that `extents WALFileLengthProvider` and won't throw IOE.

Reply:
Add some comments here?

Reply:
OK, I added comments.